[Kang Zong Xian] iP

kangzongxian opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them." - David Allen [[source](https://dansilvestre.com/productivity-quotes/)] 

DukePro frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java Programmer, you can use it to practice Java too. Here's the  ` main` code
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

kangzongxian committed:
Added Greeting, Echo and Exit Capabilities for Duke (Level-1)

kangzongxian committed:
Added Ability to store text into Array (Level-2)

kangzongxian committed:
Make Tasks able to be marked done or undone

kangzongxian committed:
Able to mark task as done or undone (Level-3)

kangzongxian committed:
Created Deadline, Event and Todo tasks (Level-4)

kangzongxian committed:
Fixed Spacing Bug

kangzongxian committed:
Added Testing (A-TextUiTesting)

kangzongxian committed:
Added Exception Handling (Level-5)

kangzongxian committed:
Added Delete Function (Level-6)

kangzongxian committed:
Created Constant Variables for Greeting and Departure with Enums

kangzongxian committed:
Fix Task Out of Range Bug

kangzongxian committed:
Implemented writing data to be saved in a File duke.txt

kangzongxian committed:
Implemented Date and Time Formats

kangzongxian committed:
Fix bug for Date Time Formats

kangzongxian committed:
Revert "Fix bug for Date Time Formats"

This reverts commit 5aa01d2c82154703a8dcd084498c52236225a97a.

kangzongxian committed:
Merge Commit, Implemented Date and Time Format

kangzongxian committed:
Fixed bug when formatting date and time and writing files (Level-7, Level-8)

kangzongxian committed:
Code Update by segmenting functionalities into more Classes (A-MoreOOP)

kangzongxian committed:
Put into duke package (A-Packages)

kangzongxian committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Required for gradle support

kangzongxian committed:
Added Gradle Support

kangzongxian committed:
Added Unit Tests (A-JUnit)

kangzongxian committed:
Created jar file for Duke (A-Jar)

kangzongxian committed:
Added JavaDocs for methods

kangzongxian committed:
Update code to adhere to a coding standard

kangzongxian committed:
Added find feature

kangzongxian committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Update code to adhere to a coding standard

kangzongxian committed:
Added find feature

kangzongxian committed:
Updated find feature

kangzongxian committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Revert "Fix bug for Date Time Formats"
  Fix bug for Date Time Formats

Neethesh26 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,29 @@
+                System.out.println("What can I do for you?");
+                break;
+            case DEPARTURE:
+                System.out.println("Bye. Hope to see you again soon ^^!");
```
Consider changing the defaults of your intelliJ so that it does not auto-indent your switch-case code to adhere to coding standards!

Neethesh26 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,29 @@
+
+    /**
+     * Constructor for the duke.Command class
+     * @param givenCommand the given command
```
I noticed this error in several of your comments, remember that there should be an empty line between your description and parameter section, also there should be punctuation at the end of the parameter description!

Neethesh26 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,186 @@
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+    private final static String FILEDESTINATION = "./src/main/duke.txt";
```
Perhaps it will be better to separate the file and destination with an underscore for readability

Neethesh26 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,186 @@
+                } else if (userCommand.equals("list")) {
+                    ui.showTaskList();
+                } else {
+                    // Get all the words the user has typed
+                    String[] words = userCommand.split(" ");
+                    // Check if user wants to mark task
+                    if (Parser.isMarkTask(words)) {
+                        int taskNumber = Integer.parseInt(words[1]);
+                        // Check if user enters a number out of range
+                        if (taskNumber < 0 || taskNumber > TaskList.taskArrayList.size()) {
+                            throw new DukeException("Number out of range!");
+                        } else {
+                            ui.markTaskDoneAndPrintOutput(taskNumber);
+                        }
+                        // Checks if user wants to unmark task
+                    } else if (Parser.isUnmarkTask(words)) {
+                        int taskNumber = Integer.parseInt(words[1]);
+                        // Check if user enters a number out of range
+                        if (taskNumber < 0 || taskNumber > TaskList.taskArrayList.size()) {
+                            throw new DukeException("Number out of range!");
+                        } else {
+                            ui.markTaskNotDoneAndPrintOutput(taskNumber);
+                        }
+                        // Check if user wants to delete a task
+                    } else if (Parser.isDeleteTask(words)) {
+                        int taskNumber = Integer.parseInt(words[1]);
+                        if (taskNumber < 0 || taskNumber >TaskList.taskArrayList.size()) {
+                            throw new DukeException("Number out of range!");
+                        } else {
+                            ui.markTaskDeletedAndPrintOutput(taskNumber);
+                        }
```
Consider separating the logical units that you have indicated well with the comments, with a blank line for better readability

Neethesh26 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,96 @@
+    // Use final variables to denote standard greetings we will say to the user
+    static final String GREETING = "Hello! I'm Duke! What can I do for you?";
+    static final String BYE = "Bye. Hope to see you again soon ^^!";
+    static final String LINEBREAK = "-------------------------";
```
Consider separating the words with a line break for readability.

Neethesh26 reviewed (commented):
Generally really well done in adhering to most code standards, other than a few minor oversights. Keep up the good work! :))

kangzongxian committed:
Fixed styling issues with CheckStyle

kangzongxian committed:
Created GUI, attempting to add FXML

kangzongxian committed:
Built GUI for Dukebot (Level-10)

kangzongxian committed:
Add method to get multiple task descriptions

kangzongxian committed:
Cleaned up Code for Duke

There were several imports that were redundant.
These were removed to make the code neater and more efficient.

Added JavaDocs for some methods that were missing.
This makes the code much more maintainable.

Code clean up makes the overall code cleaner and more maintainable.

kangzongxian committed:
Minor code clean up

kangzongxian committed:
Added assertion

kangzongxian committed:
Code Refactoring

Refactored searching for tasks into a function.

Code refactoring makes the code more maintainable.
Thus, this makes the codebase more extensible and easier to debug.

kangzongxian committed:
Used streams to make cleaner, functional code to find tasks

kangzongxian committed:
Merge pull request #1 from kangzongxian/branch-A-Assertions

Added Assertions

kangzongxian committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Added assertion

kangzongxian committed:
Merge pull request #2 from kangzongxian/branch-A-CodeQuality

Code Refactoring

kangzongxian committed:
Merge pull request #3 from kangzongxian/branch-A-Streams

Used streams to make cleaner, functional code to find tasks

kangzongxian committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Used streams to make cleaner, functional code to find tasks

kangzongxian committed:
Added Github Actions

kangzongxian committed:
Added better Search Functionality

Created a loop to loop through all the keywords.
This allows me to find partial keywords in description as well.
This makes finidng keywords more general.

kangzongxian committed:
Added comment for better readability

kangzongxian committed:
Merge pull request #4 from kangzongxian/branch-C-BetterSearch

Better Search Capabilities

kangzongxian committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:

kangzongxian committed:
Improved the GUI e.g Added coloured background

kangzongxian committed:
Added Personality to Duke (Dukity-Duke)

kangzongxian committed:
More Error Handling

kangzongxian committed:
Added more testing

kangzongxian committed:
Added Ui Picture

kangzongxian committed:
Added User Guide

kangzongxian committed:
Ceated Jar File

kangzongxian committed:
Fixed Imports in Storage

kangzongxian committed:
Fixed runtest testing

kangzongxian committed:
Fixed runtest bug

kangzongxian committed:
Fixed runtest

kangzongxian committed:
Deleted redundant code

kangzongxian committed:
Code refactoring

kangzongxian committed:
Deleted redundant code

kangzongxian committed:
Fixed deleting task bug

kangzongxian committed:
Resolve bug about saving data

kangzongxian committed:
Add auto-closure of GUI once done

kangzongxian committed:
Fix bug when saving tasks

kangzongxian committed:
Fixed DateTimeFormatting display

kangzongxian committed:
Fix storage bug

kangzongxian committed:
Add comments for better readability

kangzongxian committed:
Fix storage bug
